File: 1137-height-checker/height-checker.java
/*
By Rushabh Gala
rushabhvg
GitHub:
https://github.com/rushabhvg
StackOverflow:
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/

class Solution {
    public int heightChecker(int[] heights) {
        int n = heights.length;
        if(n==1) return 0;
        int cnt = 0;
        int[] frq = new int[101];
        int[] arr = new int[n];
        for(int i=0; i<n; i++) {
            frq[heights[i]]++;
        }
        for(int i=0; i<101; i++) {
            while(frq[i]-->0) {
                arr[cnt++] = i;
            }
        }
        cnt = 0;
        for(int i=0; i<n; i++) {
            if(arr[i]!=heights[i]) cnt++;
        }
        return cnt;
    }
}

/*
By Rushabh Gala
rushabhvg
GitHub:
https://github.com/rushabhvg
StackOverflow:
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/

File: 1137-height-checker/height-checker.rs

/*
By Rushabh Gala
rushabhvg
GitHub:
https://github.com/rushabhvg
StackOverflow:
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/

impl Solution {
    pub fn height_checker(heights: Vec<i32>) -> i32 {
        let n = heights.len();
        if n == 1 {
            return 0;
        }
        let mut cnt = 0;
        let mut frq = [0; 101];
        let mut arr = vec![0; n];
        for i in 0..n {
            frq[heights[i] as usize] += 1;
        }
        let mut idx = 0;
        for i in 0..101 {
            while frq[i] > 0 {
                arr[idx] = i as i32;
                idx += 1;
                frq[i] -= 1;
            }
        }
        cnt = 0;
        for i in 0..n {
            if arr[i] != heights[i] {
                cnt += 1;
            }
        }
        cnt
    }
}

/*
By Rushabh Gala
rushabhvg
GitHub:
https://github.com/rushabhvg
StackOverflow:
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/
